asab/web/authn/middleware.py

```diff
@@ -0,0 +1,60 @@
+import aiohttp
+
+import asab
+import asab.web.rest
+
+
+async def oauthclient_middleware_factory(request, handler, url):
```

Comment:
Can this (and every OAuth related specialization) go into a dedicated `oauth.py`?